### ISSUE TITLE
TypeError: can't compare datetime.datetime to datetime.date

### ISSUE THREAD

**Issue by nagyrobi** [franc6/ics_calendar#64]
At each update, I get the following error in the logs:

```
2022-09-26 09:55:28.723 ERROR (SyncWorker_4) [custom_components.ics_calendar.calendar] update: My Calendar: Failed to parse ICS!
Traceback (most recent call last):
  File "/config/custom_components/ics_calendar/calendar.py", line 286, in update
    self.event = self.parser.get_current_event(
  File "/config/custom_components/ics_calendar/parsers/parser_rie.py", line 109, in get_current_event
    if ParserRIE.is_event_newer(temp_end, temp_start, end, start):
  File "/config/custom_components/ics_calendar/parsers/parser_rie.py", line 128, in is_event_newer
    return start2 is None or (end2 > end and start <= start2)
TypeError: can't compare datetime.datetime to datetime.date
```

Nevertheless, the component is working and data appears in HA.

**Comment by nagyrobi** [comment]
Temporarily muted it with
```
logger:
  default: error
  logs:
    custom_components.ics_calendar.calendar: fatal
```

**Comment by franc6** [comment]
Just a quick FYI, but I do see the bug.  The main problem is that fixing the bug exposed another bug, and getting that bug fixed is proving to be a little more challenging.  I'll mark this fixed and make a release when I have both fixed.  But I don't have a lot of time to work on this right now, so it could be a while. :(

**Comment by nagyrobi** [comment]
Thanks in advance for your efforts.

**Comment by franc6** [comment]
This problem should be fixed in 3.1.3, released today.  If so, please close this issue.

**Comment by nagyrobi** [comment]
Seems OK now, thank you.
